fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt

```diff
@@ -19,28 +26,226 @@ class JunoOnboardingMapperTest {
+            ),
+        )
     }
 
     @Test
     fun showNotificationFalse_showAddWidgetFalse_pagesToDisplay_returnsSortedListOfConvertedPages_withoutNotificationPage_and_addWidgetPage() {
         val expected = listOf(defaultBrowserPageUiData, syncPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(false, false))
+        assertEquals(
+            expected,
+            unsortedAllKnownCardData.toPageUiData(
+                showNotificationPage = false,
+                showAddWidgetPage = false,
+                jexlConditions = jexlConditions,
+                func = evalFunction,
+            ),
+        )
     }
 
     @Test
     fun showNotificationFalse_showAddWidgetTrue_pagesToDisplay_returnsSortedListOfAllConvertedPages_withoutNotificationPage() {
         val expected = listOf(defaultBrowserPageUiData, addSearchWidgetPageUiData, syncPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(false, true))
+        assertEquals(
+            expected,
+            unsortedAllKnownCardData.toPageUiData(
+                showNotificationPage = false,
+                showAddWidgetPage = true,
+                jexlConditions = jexlConditions,
+                func = evalFunction,
+            ),
+        )
     }
 
     @Test
     fun showNotificationTrue_and_showAddWidgetTrue_pagesToDisplay_returnsSortedListOfConvertedPages() {
-        val expected = listOf(defaultBrowserPageUiData, addSearchWidgetPageUiData, syncPageUiData, notificationPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(true, true))
+        val expected = listOf(
+            defaultBrowserPageUiData,
+            addSearchWidgetPageUiData,
+            syncPageUiData,
+            notificationPageUiData,
+        )
+        assertEquals(
+            expected,
+            unsortedAllKnownCardData.toPageUiData(
+                showNotificationPage = true,
+                showAddWidgetPage = true,
+                jexlConditions = jexlConditions,
+                func = evalFunction,
+            ),
+        )
+    }
+
+    @Test
+    fun cardConditionsMatchJexlConditions_shouldDisplayCard_returnsConvertedPage() {
+        val jexlConditions = mapOf("ALWAYS" to "true", "NEVER" to "false")
+        val expected = listOf(defaultBrowserPageUiData)
+
+        assertEquals(
+            expected,
+            listOf(defaultBrowserCardData).toPageUiData(
+                showNotificationPage = false,
+                showAddWidgetPage = false,
+                jexlConditions = jexlConditions,
+                func = evalFunction,
+            ),
+        )
+    }
+
+    @Test
+    fun noJexlConditions_shouldDisplayCard_returnsNoPage() {
+        val jexlConditions = mapOf<String, String>()
+        val expected = emptyList<OnboardingPageUiData>()
+
+        assertEquals(
+            expected,
+            listOf(defaultBrowserCardDataNoConditions).toPageUiData(
```

Comment:
Should this use `defaultBrowserPageUiData` as the test is for empty JEXL conditions?